### ISSUE TITLE
[DL3025] False positive when single quotes used

### ISSUE THREAD

**Issue by FnTm** [hadolint/hadolint#313]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
I can use both double and single quotes within a CMD command
### Actual behavior
Using single quotes in a CMD command causes [DL3025] error to be shown, but it doesn't include any information about single quotes being bad.

### Steps to reproduce the behavior

_First order of business - thanks for making the linter!_

This error happens every time you use single quotes in a CMD directive. It's as simple as that. 
It may be that single quotes are not allowed in CMD directives, but the linter should have a separate rule for that case.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.16.2-3-g6a8dd04
```

**Dockerfile** (if relevant)

```Dockerfile
# Fails
CMD ['foo']
# Passes
CMD ["foo"]
```

**Additional environment details (OS, stack version, etc.)**

**Comment by lorenzo** [comment]
Thanks for your report, I need to investigate this further as I'm under the impression that the use of single quotes means something different

**Comment by lorenzo** [comment]
From the docker docs:

```
Note: To use a different shell, other than ‘/bin/sh’, use the exec form passing in the desired shell. For example, RUN ["/bin/bash", "-c", "echo hello"]
Note: The exec form is parsed as a JSON array, which means that you must use double-quotes (“) around words not single-quotes (‘).
```

`DL3025` actually caught a bug for you!

**Comment by lorenzo** [comment]
The same note can be found for `CMD`: https://docs.docker.com/v17.09/engine/reference/builder/#cmd

I'll close this ticket as we are correctly singling the bug. Thanks for your kind words about hadolint!

**Comment by FnTm** [comment]
Cool, I guess this is then a situation of "the more you know" 😄 Thanks for investigating @lorenzo 
But would it maybe be useful to have hadolint show a specific error about this?
Because even though it did catch the bug for me, I just went back to my original script and verify that if I use the non-array syntax of CMD - single quotes do work there. 

So I started from this:
```Dockerfile
# Works fine, notice the single quotes around the `-cp` path
CMD java -cp './*' app.main
# Works fine
CMD ["java", "-cp", "./*", "app.main"]
# Fails with [DL3025] error
CMD ["java", "-cp", './*', "app.main"]
```
My thinking there is that a linter catching this case and telling me explicitly that the issue is with the single quotes would be awesome. If I knew Haskell I'd try to come up with a PR with that myself, but that is currently quite outside of my comfort zone 🙂

**Comment by lorenzo** [comment]
You can maybe add a note in the wiki page for this linking rule?

**Comment by FnTm** [comment]
That I really can do. PR incoming!

**Comment by FnTm** [comment]
Done.

**Comment by lorenzo** [comment]
Thanks!
